projects/event-lambdas/src/lib/util.ts

```diff
@@ -14,6 +14,24 @@ import {IUserTelemetryEventWithId} from "../../../definitions/IUserTelemetryEven
 const ajv = new Ajv();
 const validateEventApiInput = ajv.compile(eventApiInputSchema);
 
+/**
+ * Determines the stage (environment) from a hostname
+ * @param hostname The hostname to extract stage from
+ * @returns The extracted stage (uppercase) or undefined if stage cannot be determined
+ */
+export const determineStageFromHostname = (hostname: string): string | undefined => {
+  if (!hostname) {
+    return undefined;
+  }
+  
+  const stageMatch = /^.*\.(?<environment>local|code)\.dev-gutools\.co\.uk$|^.*\.gutools\.co\.uk$/.exec(hostname);
+  if (stageMatch) {
+    return (stageMatch.groups?.environment || "PROD").toUpperCase();
+  }
```

Comment:
Taken from here: https://github.com/guardian/workflow-frontend/blob/main/public/tracking-pixel.js and https://github.com/guardian/tagmanager/blob/539a09bdfa6d6dfec86f85bc4871801df270ea14/public/router.js#L44-L53